### PR TITLE
SPRINT-c03 - Delete / Update Announcement.

### PR DESCRIPTION
**Pull request by Leonorjp**

### CODE REVIEW

File: src/main/java/com/criticalsoftware/AnnouncementRepository.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    public List<Announcement> findByDonorId(ObjectId donorId) {
+        return list("userDonorId", donorId);
+    }
+
+    // Find announcements by donee ID
+    public List<Announcement> findByDoneeId(ObjectId doneeId) {
+        if (doneeId != null) {
+            return list("userDoneeId", doneeId);
+        }
+        return list("userDoneeId is null");
```
Comment (jpmoreira-training) [inline src/main/java/com/criticalsoftware/AnnouncementRepository.java]: See comment on Martim's PR, please

File: src/main/java/com/criticalsoftware/AnnouncementResource.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    private AnnouncementRepository repository;
+
+    @DELETE
+    @Path("/{id}")
+    public Response delete(@PathParam("id") String id) {
+        try {
+            // Validate ObjectId
+            if (!ObjectId.isValid(id)) {
+                return Response.status(Response.Status.BAD_REQUEST).entity("Invalid announcement ID").build();
+            }
```
Comment (jpmoreira-training) [inline src/main/java/com/criticalsoftware/AnnouncementResource.java]: I would say "Invalid Id format". What do you think?

File: src/main/java/com/criticalsoftware/AnnouncementResource.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+    // Method to update an announcement by its ID
+    @PUT
+    @Path("/{id}")
+    public Response update(@PathParam("id") String id, Product product) {
+        try {
+            // Validate if the provided ID is a valid ObjectId
+            if (!ObjectId.isValid(id)) {
+                return Response.status(Response.Status.BAD_REQUEST).entity("Invalid announcement ID").build();
+            }
```
Comment (jpmoreira-training) [inline src/main/java/com/criticalsoftware/AnnouncementResource.java:48]: Same here.

File: src/main/java/com/criticalsoftware/AnnouncementResource.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+            // Convert the string ID to ObjectId
+            ObjectId objectId = new ObjectId(id);
+
```
Comment (jpmoreira-training) [inline src/main/java/com/criticalsoftware/AnnouncementResource.java]: This is well done, but I want you to try not to user PRoduct as a parameter, but a request instead.
See Martim's PR and we will talk about this.

**Review by jpmoreira-training — CHANGES_REQUESTED** [review]

File: src/main/java/com/criticalsoftware/AnnouncementResource.java (hunk 2 above)
Comment (Leonorjp) [reply src/main/java/com/criticalsoftware/AnnouncementResource.java]: Done!

File: src/main/java/com/criticalsoftware/AnnouncementResource.java (hunk 3 above)
Comment (Leonorjp) [reply src/main/java/com/criticalsoftware/AnnouncementResource.java:48]: Done!

File: src/main/java/com/criticalsoftware/Announcement.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+import io.quarkus.mongodb.panache.PanacheMongoEntity;
+import lombok.Getter;
+import lombok.Setter;
+import java.time.LocalDateTime;
+
+@Getter
+@Setter
+public class Announcement extends PanacheMongoEntity {
+
+    private String id;
```
Comment (jrromao) [inline src/main/java/com/criticalsoftware/Announcement.java:12]: As the id field is already inherited from the parent class (PanacheMongoEntity), it doesn't need to be explicitly declared in this class.

**Review by jrromao — CHANGES_REQUESTED** [review]
You can use lombok to avoid the explicit declaration of the constructors.

**Pull request merged**